Sources/MongoSwift/Counter.swift

```diff
@@ -0,0 +1,16 @@
+import Foundation
+
+/// Threadsafe counter that generates an increasing sequence of integers.
+internal class Counter {
+    private let queue = DispatchQueue(label: "counter queue")
```

Comment:
I think we should probably use a more unique identifier here, afaict this is registered with GCD so you might have conflicts.